django/db/migrations/serializer.py

```diff
@@ -244,8 +244,9 @@ def _format(self):
 
 class SetSerializer(BaseSequenceSerializer):
     def _format(self):
-        # Don't use the literal "{%s}" as it doesn't support empty set
-        return "set([%s])"
+        # When value is empty, the empty set should be serialized as "set()",
+        # not "{}" because {} is an empty dict.
+        return '{%s}' if self.value else 'set(%s)'
```

Comment:
>  I don't want to see this as a precedent for making migration files to adhere to flake8.

They are already included in runs of flake8. [See `setup.cfg`  where there is no exclude rule for `migrations`](https://github.com/django/django/blob/master/setup.cfg#L6).

This is the reason this change was originally discussed. If flake8 were to check for set literals, something would need to change with regards to migrations. Either:

* Change migrations to use set literals (this PR)
* Ignore `migrations` in flake8 (sounds like you're suggesting this alternative)

I still think there is some merit to running flake8 on migration files as they often contain written code.

Due to other reasons, the addition of flake8-comprensions won't happen for the time being, so there is no longer a need for this change. We could consider simply closing it if there is not agreement.